### PR TITLE
Topic 34 - Client Not Disconnecting on Errors or Unreachable

### LINKED ISSUE #34: OBSController doesnt stop when connection fails
```

OBSRemoteController obsRemoteController = new OBSRemoteController(url, false, pass);


if(obsRemoteController.isFailed())
{
  new StreamPiAlert("Unable to Connect to OBS", "Is it even running? Make sure the websocket plugin is installed. Check Credentials too", StreamPiAlertType.ERROR).show();
}
```

when it fails, the OBSRemoteController keeps on running, whenever you quit the application, the app doesnt stop.

### PR DESCRIPTION
**Pull request by TinaTiel**
Client now disconnects on any error (again, onError is reserved for exceptional circumstances)
Client now has configurable timeout for attempting to connect. Default is 3 seconds and must be greater than zero.

Limitation: There is an internal timeout in the Jetty websocketclient itsself that I unfortunately haven't found any documentation on, and haven't found where it could be configured in their source code. This means that if one configured a long timeout (30 seconds), atm this timeout occurs in 6 seconds latest due to the internal timeout. I've added integration tests for both cases to at least capture it.

Fixes #34 

Included as well:

- removed ObsCommunicator E2E tests / converted to test RemoteController instead; ObsCommunicator is now adequately covered by unit tests, and the ObsRemoteController E2E tests provide better value
- upgrade junit to most recent version (would explain why some features were missing, lol)

### CODE REVIEW

**Comment by TinaTiel** [comment]
These pass locally with the latest image, but the results from the test printout aren't helpfu.
I'm going to update the PR flow to provide the test report.

**Comment by TinaTiel** [comment]
Test results upload fine now. Just need to figure out why the tests pass locally but fail on the runner.

**Comment by TinaTiel** [comment]
This is now ready for review @ChristopheCVB 😆 
I think the cause of passing locally (on Windows) but failing here (Ubuntu) was due to OS differences of how connecting to nonexistent resources works. 
I've updated the test so it verifies a throwable is provided at all, and updated the implementation to guarantee the cause or its parent is always provided.

File: src/main/java/io/obswebsocket/community/client/OBSRemoteController.java [hunk 1]
```diff
@@ -67,11 +69,13 @@ public OBSRemoteController(
       ControllerLifecycleListener controllerLifecycleListener,
       String host,
       int port,
+      int connectionTimeoutSeconds,
```
Comment (ChristopheCVB) [inline src/main/java/io/obswebsocket/community/client/OBSRemoteController.java:72]: Add Javadoc param

**Review by ChristopheCVB — APPROVED** [review]

File: src/main/java/io/obswebsocket/community/client/OBSRemoteController.java (hunk 1 above)
Comment (TinaTiel) [reply src/main/java/io/obswebsocket/community/client/OBSRemoteController.java:72]: Good catch, thanks!

**Comment by TinaTiel** [comment]
Thanks for your feedback @ChristopheCVB -- I've pushed your suggestions, lemme know if all is well

**Review by ChristopheCVB — APPROVED** [review]

**Comment by rnayabed** [comment]
Great work there !!! Thank you so MUCH for this :cupid:

**Pull request merged**